### PR TITLE
pg_dump: address DROP COLUMN FIXME and add regression tests

### PR DESCRIPTION
**Pull request by doty-pivotal**
After reviewing the use of `ALTER TABLE DROP COLUMN` in the `pg_dump` code, we feel that the ordering of DDL statements in the dump do not raise any concerns with respect to vanilla table inheritance hierarchies.

In the current implementation, we need to cascade the `DROP COLUMN` to partitions, but we do not want to cascade to "normal" inherited child tables. Because the child tables are not hooked into the inheritance hierarchy until after the `DROP COLUMN` is performed, it looks like `ALTER TABLE` and `ALTER TABLE ONLY` are equivalent for vanilla inheritance.

We've added some regression tests that will hopefully catch any changes to the above assumptions. The added sql test file is run from not only `pg_regress`, but also during the standalone `pg_upgrade` cluster test. It is intended to contain corner cases specific to `pg_upgrade` where the state that needs to be tested would otherwise be destroyed during a dump and restore (such as dropped columns).

There is also a `.gitignore` touch up commit included.

### CODE REVIEW

**Comment by l-wang** [comment]
+1. Good to see we have more test coverage for pg_upgrade, so we are one step closer to test upgrade on an real world cluster which has holes in it's data files, not a brand new cluster from dump and restore. It would be great if we could add the same tests for AO/AOCO tables as well when we address this issue https://github.com/greenplum-db/gpdb/issues/6193

File: src/test/regress/sql/gp_upgrade_cornercases.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+INSERT INTO root1 VALUES (1, 2, 3), (4, 5, 6), (7, 8, 9);
+INSERT INTO child1
+VALUES (10, 100, 1000, 10000),
+       (11, 111, 1111, 11111),
+       (12, 123, 1234, 12345);
+
+ALTER TABLE root1 DROP COLUMN b;
+ALTER TABLE ONLY root1 DROP COLUMN c;
+
```
Comment (l-wang) [inline src/test/regress/sql/gp_upgrade_cornercases.sql]: Can we also add `SELECT *` in the end to verify what's being dropped and what's not?

Comment (jchampio) [reply src/test/regress/sql/gp_upgrade_cornercases.sql]: In this particular case, I'd argue that it's not the point of this file to verify that `DROP COLUMN` is doing its job -- that should be the responsibility of some other regression tests. This is just a "seed the DB with upgrade test cases" file.

Comment (l-wang) [reply src/test/regress/sql/gp_upgrade_cornercases.sql]: ah that's fair.

**Comment by jchampio** [comment]
Thanks for the reviews!
